### PR TITLE
Subcontracting Revamp

### PR DESCRIPTION
**Pull request by ryv-odoo**
See enterprise: odoo/enterprise#20350
See upgrade: odoo/upgrade#2750

### CODE REVIEW

File: addons/mrp_subcontracting_purchase/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+{
+    'name': 'Purchase and Subcontracting Management',
+    'version': '0.1',
+    'category': 'Manufacturing/Purchase',
+    'description': """
+        This bridge module adds some smart button between purchase and Subcontracting
```
Comment (amoyaux) [inline addons/mrp_subcontracting_purchase/__manifest__.py:9]: buttons

File: addons/stock/models/stock_picking.py [hunk 2]
```diff
@@ -975,10 +975,7 @@ def button_validate(self):
         return True
 
     def action_set_quantities_to_reservation(self):
-        for move in self.move_lines.filtered(lambda move: move.state in ('partially_available', 'assigned')):
-            if move.product_id.tracking == 'none':
-                for move_line in move.move_line_ids:
-                    move_line.qty_done = move_line.product_uom_qty
+        self.move_lines._set_reservation_as_done()
```
Comment (amoyaux) [inline addons/stock/models/stock_picking.py:978]: Detail but I would also rename the action name or this function name to have consistency

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo r+

**Comment by jwaes** [comment]
What is exactly the reason why you explicitly prevent by products in subcontracting ?

**Pull request closed without merging**